servicetalk-capacity-limiter-api/src/main/java/io/servicetalk/capacity/limiter/api/GradientCapacityLimiter.java

```diff
@@ -125,23 +125,29 @@ public String name() {
 
     @Override
     public Ticket tryAcquire(final Classification classification, @Nullable final ContextMap meta) {
-        int newPending;
-        int newLimit;
-
         Ticket ticket = null;
         lock.lock();
         try {
-            newLimit = (int) limit;
+            final int newLimit = (int) limit;
             if (pending < limit) {
-                newPending = ++pending;
+                final int newPending = pending + 1;
                 ticket = new DefaultTicket(this, newLimit - newPending, newPending);
+                // update the state last just in case anything throws.
+                ++pending;
```

Comment:
Doesn't look like anything can throw in this section but also doesn't hurt if we already have a `try-finally` block

Reply:
It is very unlikely but `provider.timeSource.getAsLong();` in the DefaultTicket constructor could theoretically throw. That said, if we need to guard against that we're going to have to change a lot of things.